mutation-ready.ts

```diff
@@ -0,0 +1,68 @@
+declare global {
+  interface Window {
+    WebKitMutationObserver: MutationObserver;
+  }
+}
+
+type MutationElementReady = Element & {
+  ready: boolean;
+};
+
+interface Listener {
+  selector: string;
+  callback: (element: Element) => void;
+}
+
+const listeners: Listener[] = [];
+const document: Document = window.document;
+const MutationObserver = window.MutationObserver || window.WebKitMutationObserver;
+let observer: MutationObserver | null;
+
+function check(): void {
+  // Check the DOM for elements matching a stored selector
+  for (let count = 0, length = listeners.length; count < length; count++) {
```

Comment:
Any reason we're using an old school `for` loop here instead of `for (const listener of listeners)` ?